docs/usage.md

```diff
@@ -0,0 +1,143 @@
+
+```bash
+--metadata-key cnwan.io/traffic-profile
+```
+
+will make the program only look for services whose metadata contain
+`cnwan.io/traffic-profile` and ignore all services that don't have it.
+Please note that it will only look for the *key* and will not do any
+type of filtering on the value, as this job is performed by the CNWAN
+Adaptor or whomever is in charge of handling the values.
+
+### Service Directory
+
+To connect to *Google Cloud Service Directory*, you can use the
+`servicedirectory` command. A region, project and service account path must be
+provided as flags, like so:
+
+```bash
+servicedirectory --project my-project --region us-central1 --service-account ...
+
+# With a shorter alias
+sd --project my-project --region us-central1 --service-account ...
+```
+
+Providing the service account `JSON` file is different depending on the way you
+run the project:
+
+* if you are running the binary version you can simply read
+[Binary Example](#binary-example) for a full example usage.
+
+* if you are running it as a Docker container, follow
+[Docker Usage](./docker_usage.md), as it needs some additional steps.
+
+Finally, please make sure your service account has *at least* role
+`roles/servicedirectory.viewer`. We suggest you create service account just for
+the CNWAN Reader with the aforementioned role.
+
+## Configuration File
+
+Optionally, a configuration file can be used, which can be used by providing
+its path with `--config-file` and looks like this:
+
+```yaml
+pollInterval: 20
+adaptorAPI: example.com/api:8880
+debugMode: true
+serviceDirectory:
+  metadataKey: traffic-profile
+  region: us-west2
+  project: my-project
+  serviceAccountPath: ./serv-account.json
+```
+
+The fields in the YAML file map to each CLI flag specified in the sections
+above.
+Default values will be used in case a field is not present, i.e. if
+`pollInterval` is not there, then the default value `5` will be used, as
+specified in `--help`.
+
+As for `serviceDirectory`, all fields under it are required. So you must enter
+them below it or provide them via CLI.
+
+Finally, remember that options provided via configuration file will
+**override** CLI flags. For example, if your configuration file includes
+`pollInterval: 25` but launch the program with `--interval 50`, the
+latter will be completely ignored.
+
+## Binary Example
+
+In the following example, the CNWAN Reader watches changes in
+Google Cloud Service Directory with the following requirements:
+
+* The *allowed* services have at least the `cnwan.io/traffic-profile`
+key in their metadata
+* The project is called `my-project`
+* The region is `us-west2`
+* Service account is placed inside `path/to/creds` folder
+* The name of the service account file is `serv-acc.json`
+* The endpoint of the adaptor is the default one
+(`http://localhost/cnwan/events`), so there is no need to use `--adaptor-api`
+* There is no need for a configuration file, so `--config-file` is not
```

Comment:
Instead of this, could we have an example without the config file (the one here already), and then one with the config file?